### ISSUE TITLE
chakra-ui select field calculates isMultiple wrong

### ISSUE THREAD

**Issue by julianhille** [rjsf-team/react-jsonschema-form#3848]
### Prerequisites

- [X] I have searched the existing issues
- [X] I understand that providing a [SSCCE](http://sscce.org/) example is tremendously useful to the maintainers.
- [X] I have read the [documentation](https://rjsf-team.github.io/react-jsonschema-form/docs)
- [X] Ideally, I'm providing a [sample JSFiddle](https://jsfiddle.net), [Codesandbox.io](https://codesandbox.io) or preferably a [shared playground link](https://rjsf-team.github.io/react-jsonschema-form/) demonstrating the issue.

### What theme are you using?

chakra-ui

### Version

5.12.1

### Current Behavior

A select widget acts like a multi select even if it should not.

### Expected Behavior

A select field when explicitly set to multiple=false through props should not render itself as multiple select field

### Steps To Reproduce

Here is a playground [link](https://rjsf-team.github.io/react-jsonschema-form/#eyJmb3JtRGF0YSI6eyJGcmllbmRzIjpbbnVsbF0sIkZyaWVuZCI6W10sIkZyaWVuIjpbXSwiRnJpZSI6W10sIkZyaSI6W10sIkZyIjpbXSwiRiI6W10sImZhY2lsaXRpZXMiOlt7fV0sIm9yZ2FuaXNhdGlvbiI6eyJzd2l0Y2hUb0Nob29zZSI6ZmFsc2UsImZhY2lsaXRpZXMiOlt7fV19LCJtdWx0aVNlbGVjdCI6W10sImNoZWNrYm94ZXMiOltdLCJsb2NhdGlvbiI6eyJuYW1lIjoiQW1zdGVyZGFtIiwibGF0Ijo1MiwibG9uIjo1fX0sInNjaGVtYSI6eyJwcm9wZXJ0aWVzIjp7IkZyaWVuZHMiOnsidGl0bGUiOiJGcmllbmRzIiwidHlwZSI6ImFycmF5IiwiaXRlbXMiOnsidHlwZSI6Im9iamVjdCIsInRpdGxlIjoiZnJpZW5kIiwidW5pcXVlSXRlbXMiOnRydWUsIm9uZU9mIjpbeyJwcm9wZXJ0aWVzIjp7Im5hbWUiOnsidHlwZSI6InN0cmluZyJ9fX0seyJwcm9wZXJ0aWVzIjp7ImZpcnN0bmFtZSI6eyJ0eXBlIjoibnVtYmVyIn0sImxhc3RuYW1lIjp7InR5cGUiOiJudW1iZXIifX19XX19fX0sInVpU2NoZW1hIjp7ImZhY2lsaXRpZXMiOnsiMWl0ZW1zIjp7InVpOndpZGdldCI6IlJhZGlvV2lkZ2V0In19fSwidGhlbWUiOiJjaGFrcmEtdWkiLCJsaXZlU2V0dGluZ3MiOnsic2hvd0Vycm9yTGlzdCI6InRvcCIsImV4cGVyaW1lbnRhbF9kZWZhdWx0Rm9ybVN0YXRlQmVoYXZpb3IiOnsiYXJyYXlNaW5JdGVtcyI6eyJwb3B1bGF0ZSI6InBvcHVsYXRlIiwibWVyZ2VFeHRyYURlZmF1bHRzIjpmYWxzZX0sImVtcHR5T2JqZWN0RmllbGRzIjoicG9wdWxhdGVBbGxEZWZhdWx0cyJ9fX0=`)

The issue is this [line](https://github.com/rjsf-team/react-jsonschema-form/blob/89df5cd4533e343505232ca11b1104dead82531c/packages/chakra-ui/src/SelectWidget/SelectWidget.tsx#L76): 

```
  const isMultiple = typeof multiple !== 'undefined' && Boolean(enumOptions);
```

this evaluates to true if `multiple` is set to false from multischema field [here](https://github.com/rjsf-team/react-jsonschema-form/blob/main/packages/core/src/components/fields/MultiSchemaField.tsx#L199)

### Environment

```markdown
- OS: linux fedora
- Node: 16.20.0
- npm: 8.19.4
```

**Comment by julianhille** [comment]
Would also be fixed by #3811

**Comment by heath-freenome** [comment]
Fixed by #3811
